va/va_prot.h

```diff
@@ -0,0 +1,507 @@
+ *
+ * In this example, the vaCreateBuffer is called with exec_buffer mainly becasue TEE 
+ * Communication Entrypoint buffers are CPU bound and buffer size is small enough to 
+ * have extra copy operation without impacting performance. 
+ * 
+ * \code
+ * uint32_t app_id = 0xFF;
+ * VABufferID buffer;
+ * VAProtectedSessionExecuteBuffer exec_buff = {0};
+ *
+ * exec_buff.function_id = GET_SESSION_ID;
+ * exec_buff.input.data = nullptr;
+ * exec_buff.input.data_size = 0;
+ * exec_buff.output.data = &app_id;
+ * exec_buff.output.max_data_size = sizeof(app_id);
+ * va_status = vaCreateBuffer(
+ *                 va_dpy,
+ *                 crypto_session,
+ *                 (VABufferType) VAProtectedSessionExecuteBufferType,
+ *                 sizeof(exec_buff),
+ *                 1,
+ *                 &exec_buff,
+ *                 &buffer);
+ *
+ * va_status = vaProtectedSessionExecute(va_dpy, crypto_session, buffer);
+ *
+ * vaDestroyBuffer(va_dpy, buffer);
+ * \endcode
+ *
+ * \section api_pc_attach Attach/Detach protected content session to the VA
+ * context which want to enable/disable decryption/protection
+ *
+ * Protected content session is attached to VA decode/encode/vp context to
+ * enable protected decoding/encoding/video processing per frame or entire
+ * stream. If protected session attached per frame then application has 2 
+ * options for decoding/encoding skip processing i.e. accomodating clear
+ * frames - 1. Application could do detach after each frame is processed 
+ * to process clear frame 2. Application could remains attached to decode/
+ * encode session but specify enryption byte length to 0. 
+ * The video processing does not has option #2 mainly because API does 
+ * not provide skip processing.
+ *
+ * \code
+ * vaAttachProtectedSession(va_dpy, decode_ctx, crypto_session);
+ * foreach (iteration) {
+ *     vaBeginPicture(va_dpy, decode_ctx, surface);
+ *     ...
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id1, 1);
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id2, 1);
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id_enc_param, 1);
+ *     ...
+ *     vaEndPicture(va_dpy, decode_ctx);
+ * }
+ * vaDetachProtectedSession(va_dpy, decode_ctx);
+ * \endcode
+ *
+ * or it could be frame-by-frame attaching/detaching as following:
+ *
+ * \code
+ * foreach (iteration) {
+ *     if (encrypted)
+ *         vaAttachProtectedSession(va_dpy, decode_ctx, crypto_session);
+
+ *     vaBeginPicture(va_dpy, decode_ctx, surface);
+ *     ...
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id1, 1);
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id2, 1);
+ *     vaRenderPicture(va_dpy, decode_ctx, &buf_id_enc_param, 1);
+ *     ...
+ *     vaEndPicture(va_dpy, decode_ctx);
+ *
+ *     if (encrypted)
+ *         vaDetachProtectedSession(va_dpy, decode_ctx);
+
+ *     // check encrypted variable for next frame
+ * }
+ * \endcode
+ *
+ * In above examples, the buf_id_enc_param is the buffer of type
```

Comment:
I think it's better to move this info above where you give an example. Probably you can write something like this:
```
 *     // Regular decoding buffers holding unencrypted date
 *     vaRenderPicture(va_dpy, decode_ctx, &buf_id1, 1);
 *     vaRenderPicture(va_dpy, decode_ctx, &buf_id2, 1);
 *     // Buffer holding encrypted data, i.e. VAEncryptionParameterBufferType buffer
 *     vaRenderPicture(va_dpy, decode_ctx, &buf_id_enc_param, 1);
 ``

Reply:
Updated.